### PR TITLE
Issue #1956 - Enhancements to hzn exchange org command

### PR DESCRIPTION
**Pull request by linggao**
Signed-off-by: linggao <linggao@us.ibm.com>

### CODE REVIEW

File: cli/exchange/org.go [hunk 1]
```diff
@@ -55,13 +62,16 @@ func OrgList(org, userPwCreds, theOrg string, long bool) {
 	}
 }
 
-func OrgCreate(org, userPwCreds, theOrg string, label string, desc string, min int, max int, adjust int) {
+func OrgCreate(org, userPwCreds, theOrg string, label string, desc string, tags []string, min int, max int, adjust int, maxNodes int, agbots []string) {
```
Comment (dabooz) [inline cli/exchange/org.go:65]: We dont support more than 1 agbot serving the same org.

**Review by dabooz — CHANGES_REQUESTED** [review]

File: cli/hzn.go [hunk 2]
```diff
@@ -116,19 +116,25 @@ Environment Variables:
...
 	exOrgCreateHBMax := exOrgCreateCmd.Flag("heartbeatmax", msgPrinter.Sprintf("The maximum number of seconds between agent heartbeats to the Exchange. During periods of inactivity, the agent will increase the interval between heartbeats by increments of --heartbeatadjust.")).Int()
 	exOrgCreateHBAdjust := exOrgCreateCmd.Flag("heartbeatadjust", msgPrinter.Sprintf("The number of seconds to increment the agent's heartbeat interval.")).Int()
+	exOrgCreateMaxNodes := exOrgCreateCmd.Flag("max-nodes", msgPrinter.Sprintf("The maximum number of nodes this organization is allowed to have. The default is 0 which means no limit.")).Int()
+	exOrgCreateAddToAgbot := exOrgCreateCmd.Flag("agbot", msgPrinter.Sprintf("Add the organization to this agbot's serving list. If omitted, the first agbot will be added by default. The agbot will start including the nodes, patterns and deployment policies from this organization in the search when making agreement proposals. The format is 'agbot_org/agbot_id'. This flag can be repeated to specify multiple agbots.")).Short('a').Strings()
```
[the remark this answers is not in the document]
Comment (linggao) [reply cli/hzn.go]: done. Thanks for providing accurate text.

File: cli/hzn.go [hunk 3]
```diff
@@ -116,19 +116,25 @@ Environment Variables:
 	exOrgListOrg := exOrgListCmd.Arg("org", msgPrinter.Sprintf("List this one organization.")).String()
 	exOrgListLong := exOrgListCmd.Flag("long", msgPrinter.Sprintf("Display detailed info of orgs")).Short('l').Bool()
 	exOrgCreateCmd := exOrgCmd.Command("create", msgPrinter.Sprintf("Create the organization resource in the Horizon Exchange."))
-	exOrgCreateOrg := exOrgCreateCmd.Arg("org", msgPrinter.Sprintf("Create this organization.")).Required().String()
+	exOrgCreateOrg := exOrgCreateCmd.Arg("org", msgPrinter.Sprintf("Create this organization and add it to agbot's serving list.")).Required().String()
```
[the remark this answers is not in the document]
Comment (linggao) [reply cli/hzn.go]: done

File: cli/hzn.go [hunk 4]
```diff
@@ -116,19 +116,25 @@ Environment Variables:
...
+	exOrgUpdateHBMax := exOrgUpdateCmd.Flag("heartbeatmax", msgPrinter.Sprintf("New maximum number of seconds between agent heartbeats to the Exchange. The default negative integer -1 means no change to this attribute.")).Default("-1").Int()
+	exOrgUpdateHBAdjust := exOrgUpdateCmd.Flag("heartbeatadjust", msgPrinter.Sprintf("New value for the number of seconds to increment the agent's heartbeat interval. The default negative integer -1 means no change to this attribute.")).Default("-1").Int()
+	exOrgUpdateMaxNodes := exOrgUpdateCmd.Flag("max-nodes", msgPrinter.Sprintf("The new maximum number of nodes this organization is allowed to have. The default negative integer -1 means no change.")).Default("-1").Int()
+	exOrgUpdateAddToAgbot := exOrgUpdateCmd.Flag("agbot", msgPrinter.Sprintf("Add the organization to this agbot's serving list. The agbot will start including the nodes, patterns and deployment policies from this organization in the search when making agreement proposals. The format is 'agbot_org/agbot_id'. This flag can be repeated to specify multiple agbots.")).Short('a').Strings()
```
[the remark this answers is not in the document]
Comment (linggao) [reply cli/hzn.go]: done

File: cli/exchange/org.go [hunk 5]
```diff
@@ -83,95 +93,114 @@ func OrgCreate(org, userPwCreds, theOrg string, label string, desc string, min i
...
+		if ag == "" {
+			msgPrinter.Printf("No agbot found in the Exchange.")
+			msgPrinter.Println()
+			return
+
+		}
+		agbots = []string{ag}
+	}
+
+	msgPrinter.Printf("Adding organization %v to the serving list for agbots: %v.", theOrg, agbots)
```
[the remark this answers is not in the document]
Comment (linggao) [reply cli/exchange/org.go]: done

File: cli/exchange/org.go (hunk 1 above)
Comment (linggao) [reply cli/exchange/org.go]: done

**Review by dabooz — APPROVED** [review]

**Pull request merged**